src/network.js

```diff
@@ -55,166 +57,211 @@ function networkRequest(options, oauthRequest) {
+    .then((comments) => {
+      if (!comments) {
+        return;
+      }
+
+      const lastProcessedIndex = comments.findIndex((el) => {
+        return el['data']['name'] === lastProcessedCommentId;
+      });
+      if (lastProcessedIndex !== -1) {
+        comments = comments.slice(lastProcessedIndex + 1, comments.length);
+      }
+
+      const unprocessedComments = comments
+        .map(comment => comment['data'])
+        .map(data => {
+          return {
+              'body': data['body'],
+              'author': data['author'],
+              'id': data['name'],
+              'postTitle': data['link_title'],
+              'link': data['link_permalink'] + data['id'],
+              'subreddit': data['subreddit'],
+              'timestamp' : data['created_utc']
+            }
+        });
+
+      lastProcessedCommentId = comments[0]['data']['name'];
+      return unprocessedComments;
     });
-
-  lastProcessedCommentId = content[0]['data']['name'];
-  return unprocessedComments;
 }
 
 function postComment(parentId, markdownBody) {
-  post('/api/comment', { 'parent' : parentId, 'text' : markdownBody });
+  return post('/api/comment', { 'parent' : parentId, 'text' : markdownBody });
 }
 
 function getComment(commentId) {
-  const comment = get('https://www.reddit.com/api/info.json?id=' + commentId);
-
-  if (comment.length == 0) {
-    return;
-  }
+  return get('https://www.reddit.com/api/info.json?id=' + commentId)
+    .then((comment) => {
+      if (comment.length == 0) {
+        return;
+      }
 
-  const data = comment[0]['data'];
+      const data = comment[0]['data'];
 
-  return {
-    'body': data['body'],
-    'author': data['author'],
-    'id': data['name'],
-    'link_id' : data['link_id'],
-    'postTitle': '', // api/info does not return a value for postTitle but this property is required by shouldConvertComment
-    'subreddit': data['subreddit'],
-    'timestamp' : data['created_utc']
-  }
+      return {
+        'body': data['body'],
+        'author': data['author'],
+        'id': data['name'],
+        'link_id' : data['link_id'],
+        'postTitle': '', // api/info does not return a value for postTitle but this property is required by shouldConvertComment
+        'subreddit': data['subreddit'],
+        'timestamp' : data['created_utc']
+      }
+    });
 }
 
 function editComment(commentId, markdownBody) {
-  post('/api/editusertext', { 'thing_id' : commentId, 'text' : markdownBody });
+  return post('/api/editusertext', { 'thing_id' : commentId, 'text' : markdownBody });
 }
 
 function getCommentReplies(linkId, commentId) {
-  const replies = get('https://www.reddit.com/api/morechildren.json?api_type=json&link_id=' + linkId + '&children=' + commentId);
+  return get('https://www.reddit.com/api/morechildren.json?api_type=json&link_id=' + linkId + '&children=' + commentId)
+    .then((replies) => {
+      if (replies.length === 0) {
```

Comment:
The diff is a little off, but this early out conditional was previously 
```js
if (! replies.length === 0) {
```
, which seems like you would _not_ want to early out when you _do_ have replies.

If that was intentional though, this change should be reverted.